skimage/util/tests/test_map_array.py

```diff
@@ -1,9 +1,26 @@
 import numpy as np
+import pytest
 from skimage.util._map_array import map_array, ArrayMap
 
 from skimage._shared import testing
 
 
+@pytest.mark.parametrize(
+    "dtype",
+    [np.uint8, np.uint16, np.uint32, np.uint64, np.int8, np.int16, np.int32, np.int64],
+)
+def test_map_array_simple(dtype):
+    input_arr = np.array([0, 2, 0, 3, 4, 5, 0], dtype=dtype)
+    input_vals = np.array([1, 2, 3, 4, 6], dtype=dtype)[::-1]
+    output_vals = np.array([6, 7, 8, 9, 10], dtype=dtype)[::-1]
+    desired = np.array([0, 7, 0, 8, 9, 0, 0], dtype=dtype)
+    result = map_array(
+        input_arr=input_arr, input_vals=input_vals, output_vals=output_vals
+    )
+    np.testing.assert_array_equal(result, desired)
+    assert result.dtype == dtype
```

Comment:
A cool feature of map_array is that the in dtype and out dtype don't have to match. So maybe change this to:

```suggestion
@pytest.mark.parametrize(
    "dtype_in,dtype_out",
    itertools.product(
        [np.uint8, np.uint16, np.uint32, np.uint64, np.int8, np.int16, np.int32, np.int64],
        repeat=2,
    )
)
def test_map_array_simple(dtype_in, dtype_out):
    input_arr = np.array([0, 2, 0, 3, 4, 5, 0], dtype=dtype_in)
    input_vals = np.array([1, 2, 3, 4, 6], dtype=dtype_in)[::-1]
    output_vals = np.array([6, 7, 8, 9, 10], dtype=dtype_out)[::-1]
    desired = np.array([0, 7, 0, 8, 9, 0, 0], dtype=dtype_out)
    result = map_array(
        input_arr=input_arr, input_vals=input_vals, output_vals=output_vals
    )
    np.testing.assert_array_equal(result, desired)
    assert result.dtype == dtype_out
```

Although in fact maybe output should also include float types... 😅

Reply:
Addressed in https://github.com/scikit-image/scikit-image/pull/7266/commits/aecf6f5f6200474225e0f7de6fcf41f6bc73b386.